src/Features/Core/Portable/Diagnostics/Analyzers/IDocumentDiagnosticAnalyzer.cs

```diff
@@ -12,10 +11,14 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal abstract class DocumentDiagnosticAnalyzer : DiagnosticAnalyzer
     {
+        // REVIEW: why DocumentDiagnosticAnalyzer doesn't have span based analysis?
```

Comment:
span based analyzer means caller ask span to analyze not analyzer itself determine span to analyze. so if an analyzer has ability to return diagnostics based on given span, it can claim it is span based (or my wording might be wrong that confused you)